libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookEvents/FacebookEntry.cs

```diff
@@ -23,18 +23,21 @@ public class FacebookEntry
         /// Gets or sets the messaging list.
         /// </summary>
         /// <value>List containing one messaging object. Note that even though this is an enumerable, it will only contain one object.</value>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "CA2227:Collection properties should be read only", Justification = "it needs to be set in ActivityToFacebook method")]
         public List<FacebookMessage> Messaging { get; set; }
 
         /// <summary>
         /// Gets or sets the changes list.
         /// </summary>
         /// <value>List containing the list of changes.</value>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "CA2227:Collection properties should be read only", Justification = "it needs to be set in ActivityToFacebook method")]
         public List<FacebookMessage> Changes { get; set; }
 
         /// <summary>
         /// Gets or sets the standby messages list.
         /// </summary>
         /// <value>List containing the messages sent while in standby mode.</value>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "CA2227:Collection properties should be read only", Justification = "it needs to be set in ActivityToFacebook method")]
```

Comment:
Same as above.

Then you can refactor the code in FacebookAdapter to iterate over the list rather than checking for null.